OpenRA.Mods.Common/Traits/CombatDebugOverlay.cs

```diff
@@ -70,6 +72,22 @@ void IRenderAboveWorld.RenderAboveWorld(Actor self, WorldRenderer wr)
 				TargetLineRenderable.DrawTargetMarker(wr, hc, hb);
 			}
 
+			var tc = Color.Lime;
+			var enabledPositions = targetablePositions.Where(Exts.IsTraitEnabled);
+			foreach (var t in enabledPositions)
+			{
+				var positions = enabledPositions.SelectMany(tp => tp.TargetablePositions(self));
```

Comment:
I noticed myself that the `foreach` above is bogus/redundant, because this line already selects all `TargetablePositions` across all active provider-traits, so this would draw all TPs as many times as there are traits. Not a big deal in practice, but wastes some performance.
Will push a fix shortly.